src/main/java/services/ClusterService.java

```diff
@@ -306,6 +320,26 @@ default void inputDetail() {
           */
          @In ClusterUpgradeAction upgradeAction();
 
+         /**
+          * Explicitly set the upgrade correlation identifier.  Use to correlate events
+          * detailing the cluster upgrade to the upgrade itself.  If not specificed, the
+          * correlation id from `Correlation-Id` http header will be used.
+          *
+          * @author Scott Dickerson <sdickers@redhat.com>
+          * @date 17 Mar 2022
+          * @since 4.5.0
+          */
+         @In String correlationId();
```

Comment:
Consider renaming to upgradeCorrelationId

Reply:
I'm ok with leaving the `upgrade` part off the of the service parameter name.